### DISCUSSION TITLE
定义数组属性，哪怕没有添加，数据中也会是一个空数组，可以为undefined吗

### DISCUSSION THREAD

**Discussion by superchenwb** [alibaba/formily#2142] [Q&A 1.x]
这是官网的案例
https://codesandbox.io/s/d0hvu

假设定义了数组属性，什么操作都没做，直接点击提交，表单数组中string_array会是一个空数组，
有什么办法能在没有添加值时，values中不存在，或者值为undefined吗？

**Comment by janryWang** [comment]
不行，默认如果为空的话，自定义组件每次都需要判断空再渲染，开发体验很不好

**Reply by superchenwb** [reply]
我使用formily做组件的属性面板时，有一些组件的属性是受控属性，像Collapse的属性activeKey，如果出现activeKey为[]，但是我的本意是让它为undefined的

我有什么方法能够自己进行处理吗？比如说自己写一个ArrayItems

**Reply by janryWang** [reply]
不指定array就行

**Reply by superchenwb** [reply]
不好意思，之前没有及时回复。
不指定array虽然不影响渲染，但是会导致新增按钮点击后无效，请问还有其他的方案吗
